src/main/scala/edu/ie3/simona/main/RunSimonaStandalone.scala

```diff
@@ -56,9 +62,38 @@ object RunSimonaStandalone extends RunSimona[SimonaStandaloneSetup] {
       case SimonaEnded(successful) =>
         simonaSim.terminate()
 
+        val config = SimonaConfig(simonaSetup.typeSafeConfig).simona.output
+
+        config.sink.csv.map(_.zipFiles).foreach { zipFiles =>
+          if (zipFiles) {
+            val rawOutputPath =
+              Path.of(simonaSetup.resultFileHierarchy.rawOutputDataDir)
+
+            rawOutputPath.toFile.listFiles().foreach { file =>
+              val fileName = file.getName
+              val archiveName = fileName.replace(".csv", "")
+              val filePath = rawOutputPath.resolve(fileName)
+
+              val compressFuture =
+                FileIOUtils
+                  .compressFile(filePath, rawOutputPath.resolve(archiveName))
+                  .asScala
+              compressFuture.onComplete {
+                case Success(_) =>
+                  FileIOUtils.deleteRecursively(filePath)
+                case Failure(exception) =>
+                  logger.error(
+                    s"Compression of output file to '$archiveName' has failed. Keep raw data.",
+                    exception,
+                  )
+              }
+              Await.ready(compressFuture, compressTimeoutDuration)
+            }
+          }
+        }
+
```

Comment:
Sorry, only looked into this now. Maybe this would've fit better inside `ResultEventListener` while handling `StopTimeout` (i.e. right before the listener stops). This might look pedantic, but I think it's better to keep functionality where it belongs structure-wise. Will provide a suggestion later.